sqlglot/parser.py

```diff
@@ -1123,6 +1102,20 @@ def _parse_fallback(self, no=False) -> exp.Expression:
             exp.FallbackProperty, no=no, protection=self._match_text_seq("PROTECTION")
         )
 
+    def _parse_with_property(
```

Comment:
This was introduced because previously we were matching both `"WITH"` and `TokenType.WITH`, so we somehow need to disambiguate.